### PR TITLE
Build image and push to GHCR

### PR DESCRIPTION
**Pull request by EyeCantCU**
This builds an image by passing the Dockerfile to buildah and submits it to GHCR

Note: Requires generating a key pair with cosign

### Pre-submission checklist:

*Please check each of these after submitting your pull request:*

* [ ] Are you only including a `repo_url` if your project is 100% open source? If so, you need to pick the single best GitHub repository for your project, not a GitHub organization.
* [ ] Is your project closed source or, if it is open source, does your project have at least 300 GitHub stars?
* [ ] Have you picked the single best (existing) category for your project?
* [ ] Does it follow the other guidelines from the [new entries](https://github.com/cncf/landscape#new-entries) section?
* [ ] Have you added your SVG to `hosted_logos` and referenced it there?
* [ ] Does your logo clearly state the name of the project/product and follow the other logo [guidelines](https://github.com/cncf/landscape#logos)?
* [ ] Does your project/product name match the text on the logo?
* [ ] Have you verified that the Crunchbase data for your organization is correct (including headquarters and LinkedIn)?
* [ ] ~30 minutes after opening the pull request, the CNCF-Bot will post the URL for your staging server. Have you confirmed that it looks good to you and then added a comment to the PR saying "LGTM"?

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+      contents: read
+      packages: write
+      id-token: write
+    strategy:
+      fail-fast: false
+      matrix:
+        image_name: [landscape]
+    steps:
+      - name: Maximize build space
+        uses: ublue-os/remove-unwanted-software@v6
```
Comment (castrojo) [inline .github/workflows/build.yml:36]:
```suggestion
```
This can go, CNCF has larger builders and we shouldn't add this dep here.

**Review by castrojo — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml (hunk 1 above)
Comment (EyeCantCU) [reply .github/workflows/build.yml]: Awesome! The steps line is still needed. I'll get that fixed :)

**Comment by jeefy** [comment]
Last commit needs to be signed, otherwise generally LGTM!

**Comment by EyeCantCU** [comment]
> Last commit needs to be signed, otherwise generally LGTM!

Will handle that right now :)

**Comment by castrojo** [comment]
I think we'll need this for landscape2, RJ I'll follow up with you on this one separately.

**Comment by tegioz** [comment]
The landscape2 CLI tool is also distributed in a [container image](https://github.com/cncf/landscape2?tab=readme-ov-file#container-image) @castrojo. All the [build and deployment workflows for all landscapes we host](https://github.com/cncf/landscape2-sites/tree/main/.github/workflows) rely on it actually 🙂

**Pull request closed without merging**